src/include/api/vamana_index.h

```diff
@@ -252,6 +234,28 @@ class IndexVamana {
     index_->write_index(ctx, group_uri, timestamp, storage_version);
   }
 
+  static void clear_history(
+      const tiledb::Context& ctx,
+      const std::string& group_uri,
+      uint64_t timestamp) {
+    tiledb_datatype_t feature_datatype{TILEDB_ANY};
+    tiledb_datatype_t id_datatype{TILEDB_ANY};
+    tiledb_datatype_t adjacency_row_index_datatype{TILEDB_ANY};
+    read_types(
+        ctx,
+        group_uri,
+        &feature_datatype,
+        &id_datatype,
+        &adjacency_row_index_datatype);
+
+    auto type =
+        std::tuple{feature_datatype, id_datatype, adjacency_row_index_datatype};
+    if (uri_dispatch_table.find(type) == uri_dispatch_table.end()) {
```

Comment:
TODO(paris): This should be `clear_history_dispatch_table` - will fix it in the follow-up so I don't need to re-run CI for this PR.